tests/accessibility/pa11y.js

```diff
@@ -0,0 +1,58 @@
+const fs = require('fs')
+const path = require('path')
+const archiver = require('archiver')
+const pa11y = require('pa11y')
+const htmlReporter = require('pa11y-reporter-html')
+
+const reportsPath = path.join(__dirname, 'reports.zip')
+
+let archive
+
+const tests = [
+  // pa11y('http://localhost:3000/form-example/3') <- add URL to test here
+]
+
+generateReportsForFailedTests()
+
+async function generateReportsForFailedTests() {
+  clearReports()
+
+  for (let test of tests) {
+    let results = await test
+
+    if (results.issues.length > 0) {
+      await createReport(results)
+    }
+  }
+
+  if (archive) {
+    archive.finalize()
+    throw 'Accessibility tests FAILED' // Fail build
+  }
+}
+
+function clearReports() {
+  if (fs.existsSync(reportsPath)) {
+    fs.unlinkSync(reportsPath)
+  }
+}
+
+async function createReport(results) {
+  if (!archive) {
+    initArchive()
+  }
+
+  let htmlResults = await htmlReporter.results(results)
+  archive.append(htmlResults, { name: generateFileName(results.pageUrl) })
+}
+
+function initArchive() {
+  archive = archiver('zip')
```

Comment:
is there a reason we zip the html files up?

Reply:
Requested by ops so that we can download a single file rather than multiple, as I believe this requires manual configuration to make this available as an artefact